### DISCUSSION TITLE
Firmware Upload Troobleshooting user guide.

### DISCUSSION THREAD

**Discussion by PPAC37** [MarginallyClever/Makelangelo-software#424] [Q&A]
As the firmware download hides multiple prerequisites and possible problems in its accomplishment, it will be good to have a wiki page that the user is redirected to.

This page would contain information like this.

ArduinoIDE download url, avrdude, XLoader, ...

download links for a Windows driver CH340 / CH341 (normally onboard by ArduinoIDE) to have a COM port on a USB connection.

Verification methods that the Windows driver is properly installed (check in the device manager that when you connect the USB cable between the PC and the motherboard of the polargraph machine, it finds a COM port with a number and not an unknown device or with a "!" icon.)

Often when you plug in the USB cable, the + 5V from the USB allows you to power the microcontroller of the card (but not the motors which require more currents) and we often find a see several led indicators which s' light up. (if not then check the USB cables, the connector on the polargraph card (the micro USB can easily be partially torn off / unbonded ...)

Note on the choice of the USB cable to use (a data transfer cable, (and not only a charge because otherwise the DATA + and DATA- wires are missing in the USB cable and they are essential for data transfer) that We must avoid damaged USB cables, too long, unshielded and from time to time poor quality cables may need to be hacked (forward or backward) in the USB connector for contact to be made.)

avoid connecting to a USB 3.0 port on the PC but rather use a USB 2.0 or USB 1 port

And the list of possible problem / difficulty is not finished.
...

This will help direct users new to USB connection and firmware transfer history to a checklist and knowledge base.

**Comment by PPAC37** [comment]
+ link to avrdude documentation , +link to linux serial port : how to configure my user to be in the users group that can use the serial port (normaly done by ArduinoIDE when installed)  ...

**Comment by PPAC37** [comment]
Explain that a COM port can only be used by one application at a time.
For example, Ultimaker Cura tries to automatically connect to available COM ports. Ultimaker cura must be closed for the COM port to be usable. ....

**Comment by i-make-robots** [comment]
mention OSX security issues prevent users from reading files from disk.
